realm/realm-library/src/main/java/io/realm/RealmObject.java

```diff
@@ -176,11 +180,16 @@ public final boolean load() {
     boolean onCompleted() {
         try {
             Long handoverResult = pendingQuery.get();// make the query blocking
-            // this may fail with BadVersionException if the caller and/or the worker thread
-            // are not in sync (same shared_group version).
-            // REALM_COMPLETED_ASYNC_FIND_FIRST will be fired by the worker thread
-            // this should handle more complex use cases like retry, ignore etc
-            onCompleted(handoverResult);
+            if (handoverResult != 0) { // no row found
```

Comment:
In `HandlerController.completedAsyncRealmObject()`, `onCompleted(long)` and `notifyChangeListeners()` are always called even if `pointer == 0`.
Why the codes are different?